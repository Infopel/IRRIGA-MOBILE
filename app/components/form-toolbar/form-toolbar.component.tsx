import { MaterialIcons } from "@expo/vector-icons"
import { observer } from "mobx-react-lite"
import { Box, Heading, HStack, Icon, IconButton, Pressable, StatusBar, Text, VStack } from "native-base"
import * as React from "react"
import { StyleProp, ViewStyle } from "react-native"
import { goBack } from "navigators"
import { FormToolbarStyles as styles } from "./form-toolbar.styles"

export interface FormToolbarProps {
  /**
   * An optional style override useful for padding & margin.
   */
  style?: StyleProp<ViewStyle>
  totalPages: number
  pageTitle: string
  pageNumber: number
  error?:string
  changePage: (page: number) => void
}

/**
 * Describe your component here
 */
export const FormToolbar = observer(function (props: FormToolbarProps) {
  const { style, changePage, totalPages, pageTitle, pageNumber, error } = props
  const containerStyle = {
    ...styles.container,
    ...(style && typeof style === "object" ? style : {}),
  }

  const listOfNumbers = React.useCallback(() => {
    const list: number[] = []
    if (!totalPages) return list
    for (let i = 0; i < totalPages; i++) {
      list[i] = 1 + i
    }
    return list
  }, [])

  return (
    <VStack>
      <StatusBar />
      <HStack alignItems={"flex-start"} style={containerStyle} bg="secondary.400" space={2}>
        <IconButton
          _icon={{
            color: "white",
            size: "md",
          }}
          p={"1"}
          onPress={goBack}
          icon={<Icon as={MaterialIcons} name="arrow-back" />}
        />
        {/* <Box
          borderRadius="full"
          borderColor="white"
          _text={{
            color: "white",
            fontWeight: "bold",
            fontSize: "20",
          }}
          justifyContent="center"
          borderWidth={3}
          px={5}
        >
          {pageInfo?.pageNumber}
        </Box> */}
        <VStack space={2}>
          <Heading size={"md"} color={"white"}>
            {pageTitle}
          </Heading>
          <HStack>
            {listOfNumbers().map((i) => (
              <Pressable key={i} onPress={changePage.bind(this, i - 1)}>
                <Box
                  borderRadius="full"
                  bg={pageNumber === i ? "white" : "secondary.100"}
                  p={2}
                  mx="1"
                />
              </Pressable>
            ))}
          </HStack>
        </VStack>
      </HStack>
      {error && (
        <HStack
          _light={{
            bg: "light.200",
          }}
          _dark={{
            bg: "dark.200",
          }}
          p={2}
          justifyContent={"center"}
          w="full"
        >
          <Text color="danger.500" textAlign={"center"}>
            {error}
          </Text>
        </HStack>
      )}
    </VStack>
  )
})
