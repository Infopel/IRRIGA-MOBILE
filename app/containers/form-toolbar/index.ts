export { FormToolbarContainer } from './form-toolbar.container'