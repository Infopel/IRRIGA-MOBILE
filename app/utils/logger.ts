const formatMessages = (messages: any) => {
  const [first, ...other] = messages
  return [typeof first === "string" ? first : first, ...other]
}

class Logger {
  silent: boolean = false
  private console = __DEV__ ? console.tron : console

  log(...messages: any[]): void {
    !this.silent && this.console.log(messages)
  }

  warn(...messages: any[]): void {
    !this.silent && this.console.warn(formatMessages(messages))
  }

  debug(message: any, isImportant: boolean = false){
    if(__DEV__){
      this.console.debug(message, isImportant)
    }else {
      this.console.debug(message)
    }
  }

  error(e: Error | unknown, message?: string): void {
    if (e instanceof Error) {
      !this.silent && this.console.error(e?.message ?? message, e?.stack)
    } else !this.silent && this.console.error(e, "Unknown error" ?? message)
  }

  silence(): void {
    this.silent = true
  }
}

export default new Logger()
