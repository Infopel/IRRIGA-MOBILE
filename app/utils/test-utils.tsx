import DatabaseProvider from "@nozbe/watermelondb/DatabaseProvider"
import { render } from "@testing-library/react-native"
import React from "react"
import { database } from "storage"
import { NativeBaseProvider as ThemeProvider } from "theme"
export type AllTheProvidersProps = { children: React.ReactNode }
function AllTheProviders({ children }: AllTheProvidersProps) {
  return (
    <DatabaseProvider database={database}>
      <ThemeProvider>{children}</ThemeProvider>
    </DatabaseProvider>
  )
}

type RenderOptions = {
  wrapper?: React.ComponentType<any>;
  createNodeMock?: (element: React.ReactElement) => any;
}

function customRender<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

// re-export everything
export * from "@testing-library/react-native"
// override render method
export { customRender as render }
