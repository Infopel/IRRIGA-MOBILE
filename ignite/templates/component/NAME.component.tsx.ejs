import * as React from "react"
import { StyleProp, TextStyle, View, ViewStyle } from "react-native"
import { color, typography } from "../../theme"
import { Text } from "../../components"
import {<%= props.pascalCaseName %>Styles as styles} from "./<%= props.kebabCaseName %>.styles"

export interface <%= props.pascalCaseName %>Props {
  /**
   * An optional style override useful for padding & margin.
   */
  style?: StyleProp<ViewStyle>
}

/**
 * Describe your component here
 */
export const <%= props.pascalCaseName %> = function <%= props.pascalCaseName %>(props: <%= props.pascalCaseName %>Props) {
  const { style } = props
  const containerStyle = {
    ...styles.container,
    ...(style && typeof style === "object" ? style : {}),
  }

  return (
    <View style={containerStyle}>
      <Text style={styles.text}>Hello</Text>
    </View>
  )
}
