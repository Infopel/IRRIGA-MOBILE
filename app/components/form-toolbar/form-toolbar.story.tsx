import * as React from "react"
import { storiesOf } from "@storybook/react-native"
import { StoryScreen, Story, UseCase } from "../../../storybook/views"
import { color } from "../../theme"
import { FormToolbar } from "./form-toolbar.component"
import { NativeBaseProvider, theme } from "native-base"

storiesOf("FormToolbar", module)
  .addDecorator((fn) => <StoryScreen>{fn()}</StoryScreen>)
  .add("Default", () => (
    <Story>
      <UseCase text="Primary" usage="The primary.">
        <FormToolbar
          {...{
            changePage: () => {},
            pageNumber: 1,
            pageTitle: "Form Toolbar",
            totalPages: 8,
          }}
        />
      </UseCase>
    </Story>
  ))
