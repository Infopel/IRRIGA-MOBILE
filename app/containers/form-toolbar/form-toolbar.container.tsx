import { FormToolbarSkeleton } from "components/form-toolbar-skeleton/form-toolbar-skeleton.component"
import { observer } from "mobx-react-lite"
import { useStores } from "models"
import * as React from "react"
import { StyleProp, ViewStyle } from "react-native"
import { FormToolbar, Text } from "../../components"

export interface FormToolbarProps {
  /**
   * An optional style override useful for padding & margin.
   */
  style?: StyleProp<ViewStyle>
}

/**
 * Describe your component here
 */
export const FormToolbarContainer = observer((props: FormToolbarProps) => {
  const {
    formStore: { isRequestingForm, changePage, pageInfo, error },
  } = useStores()

  if (isRequestingForm) {
    return <FormToolbarSkeleton/>
  }

  if (!pageInfo) return null
  return (
    <>
      <FormToolbar {...{ ...pageInfo, changePage, error }} />
    </>
  )
})
