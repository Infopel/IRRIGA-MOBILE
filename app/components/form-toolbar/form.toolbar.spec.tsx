import React from 'react'
import { fireEvent, render, screen } from "utils"
import { FormToolbar } from './form-toolbar.component'
describe("<FormToolbar>", () => {
    it("renders", () => {

        const changePage = jest.fn()
        const pageNumber = 5
        const pageTitle = "Simples titulo"
        const disabledPages: number[] = []
        const totalPages = 8
        render(<FormToolbar {...{ changePage, disabledPages, pageNumber, pageTitle, totalPages }} />)
        expect(screen.getAllByRole('button').length).toBe(totalPages + 1)
        expect(screen.getByAccessibilityState({ selected: true })).toBeTruthy()
    })
    it("should renders 2 disabled pages", () => {

        const changePage = jest.fn()
        const pageNumber = 5
        const pageTitle = "Simples titulo"
        const disabledPages = [2, 3]
        const totalPages = 8
        render(<FormToolbar {...{ changePage, disabledPages, pageNumber, pageTitle, totalPages }} />)
        expect(screen.getAllByRole('button').length).toBe(totalPages + 1)
        expect(screen.getByAccessibilityState({ selected: true })).toBeTruthy()
        expect(screen.getAllByAccessibilityState({ disabled: true }).length).toBe(2)
    })

    it("should show disabled content", () => {
        const changePage = jest.fn()
        const pageNumber = 5
        const pageTitle = "Simples titulo"
        const totalPages = 8
        const disabledPages = [2, 3]
        render(<FormToolbar {...{ changePage, disabledPages, pageNumber, pageTitle, totalPages }} />)
        expect(screen.getAllByAccessibilityState({ disabled: true }).length).toBe(disabledPages.length)
        expect(screen.toJSON()).toMatchSnapshot()

    })

    it("renders - snapshot", () => {

        const changePage = jest.fn()
        const pageNumber = 5
        const pageTitle = "Simples titulo"
        const totalPages = 8
        const disabledPages: number[] = []
        render(<FormToolbar {...{ changePage, disabledPages, pageNumber, pageTitle, totalPages }} />)
        expect(screen.toJSON()).toMatchSnapshot()
    })


    it("should change page be called", () => {
        const changePage = jest.fn()
        const pageNumber = 5
        const pageTitle = "Simples titulo"
        const totalPages = 8
        const disabledPages = [2, 3]
        render(<FormToolbar {...{ changePage, disabledPages, pageNumber, pageTitle, totalPages }} />)
        screen.debug()
        const btn = screen.getAllByRole('button')[2]
        expect(btn).toBeTruthy()
        fireEvent.press(btn)

        expect(changePage).toBeCalled()
    })
})