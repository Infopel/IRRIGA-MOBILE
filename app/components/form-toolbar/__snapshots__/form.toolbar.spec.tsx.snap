// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormToolbar> renders - snapshot 1`] = `
<View
  dataSet={Object {}}
  style={Object {}}
>
  <View
    dataSet={Object {}}
    style={Object {}}
  />
  <View
    dataSet={Object {}}
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "#35afe3",
          "flexDirection": "row",
        },
        Object {
          "padding": 16,
        },
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      colorScheme="primary"
      dataSet={Object {}}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onHoverIn={[Function]}
      onHoverOut={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 4,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 4,
        }
      }
    >
      <Text />
    </View>
    <View
      dataSet={Object {}}
      style={
        Object {
          "width": 8,
        }
      }
    />
    <View
      dataSet={Object {}}
      style={Object {}}
    >
      <Text
        dataSet={Object {}}
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#FFFFFF",
            "fontFamily": undefined,
            "fontSize": 20,
            "fontStyle": "normal",
            "fontWeight": "700",
            "letterSpacing": 0,
            "lineHeight": 25,
            "textDecorationLine": undefined,
          }
        }
      >
        Simples titulo
      </Text>
      <View
        dataSet={Object {}}
        style={
          Object {
            "height": 8,
          }
        }
      />
      <View
        dataSet={Object {}}
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onHoverIn={[Function]}
          onHoverOut={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={Object {}}
        >
          <View
            accessibilityState={
              Object {
                "disabled": false,
                "selected": false,
              }
            }
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": "#b5e5fa",
                "borderRadius": 9999,
                "marginLeft": 4,
                "marginRight": 4,
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
              }
            }
          />
        </View>
        <View
          dataSet={Object {}}
          style={Object {}}
        />
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onHoverIn={[Function]}
          onHoverOut={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={Object {}}
        >
          <View
            accessibilityState={
              Object {
                "disabled": false,
                "selected": false,
              }
            }
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": "#b5e5fa",
                "borderRadius": 9999,
                "marginLeft": 4,
                "marginRight": 4,
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
              }
            }
          />
        </View>
        <View
          dataSet={Object {}}
          style={Object {}}
        />
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onHoverIn={[Function]}
          onHoverOut={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={Object {}}
        >
          <View
            accessibilityState={
              Object {
                "disabled": false,
                "selected": false,
              }
            }
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": "#b5e5fa",
                "borderRadius": 9999,
                "marginLeft": 4,
                "marginRight": 4,
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
              }
            }
          />
        </View>
        <View
          dataSet={Object {}}
          style={Object {}}
        />
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onHoverIn={[Function]}
          onHoverOut={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={Object {}}
        >
          <View
            accessibilityState={
              Object {
                "disabled": false,
                "selected": false,
              }
            }
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": "#b5e5fa",
                "borderRadius": 9999,
                "marginLeft": 4,
                "marginRight": 4,
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
              }
            }
          />
        </View>
        <View
          dataSet={Object {}}
          style={Object {}}
        />
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onHoverIn={[Function]}
          onHoverOut={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={Object {}}
        >
          <View
            accessibilityState={
              Object {
                "disabled": false,
                "selected": true,
              }
            }
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 9999,
                "marginLeft": 4,
                "marginRight": 4,
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
              }
            }
          />
        </View>
        <View
          dataSet={Object {}}
          style={Object {}}
        />
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onHoverIn={[Function]}
          onHoverOut={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={Object {}}
        >
          <View
            accessibilityState={
              Object {
                "disabled": false,
                "selected": false,
              }
            }
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": "#b5e5fa",
                "borderRadius": 9999,
                "marginLeft": 4,
                "marginRight": 4,
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
              }
            }
          />
        </View>
        <View
          dataSet={Object {}}
          style={Object {}}
        />
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onHoverIn={[Function]}
          onHoverOut={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={Object {}}
        >
          <View
            accessibilityState={
              Object {
                "disabled": false,
                "selected": false,
              }
            }
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": "#b5e5fa",
                "borderRadius": 9999,
                "marginLeft": 4,
                "marginRight": 4,
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
              }
            }
          />
        </View>
        <View
          dataSet={Object {}}
          style={Object {}}
        />
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onHoverIn={[Function]}
          onHoverOut={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={Object {}}
        >
          <View
            accessibilityState={
              Object {
                "disabled": false,
                "selected": false,
              }
            }
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": "#b5e5fa",
                "borderRadius": 9999,
                "marginLeft": 4,
                "marginRight": 4,
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`<FormToolbar> should show disabled content 1`] = `
<View
  dataSet={Object {}}
  style={Object {}}
>
  <View
    dataSet={Object {}}
    style={Object {}}
  />
  <View
    dataSet={Object {}}
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "#35afe3",
          "flexDirection": "row",
        },
        Object {
          "padding": 16,
        },
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      colorScheme="primary"
      dataSet={Object {}}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onHoverIn={[Function]}
      onHoverOut={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 4,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 4,
        }
      }
    >
      <Text />
    </View>
    <View
      dataSet={Object {}}
      style={
        Object {
          "width": 8,
        }
      }
    />
    <View
      dataSet={Object {}}
      style={Object {}}
    >
      <Text
        dataSet={Object {}}
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#FFFFFF",
            "fontFamily": undefined,
            "fontSize": 20,
            "fontStyle": "normal",
            "fontWeight": "700",
            "letterSpacing": 0,
            "lineHeight": 25,
            "textDecorationLine": undefined,
          }
        }
      >
        Simples titulo
      </Text>
      <View
        dataSet={Object {}}
        style={
          Object {
            "height": 8,
          }
        }
      />
      <View
        dataSet={Object {}}
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onHoverIn={[Function]}
          onHoverOut={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={Object {}}
        >
          <View
            accessibilityState={
              Object {
                "disabled": false,
                "selected": false,
              }
            }
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": "#b5e5fa",
                "borderRadius": 9999,
                "marginLeft": 4,
                "marginRight": 4,
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
              }
            }
          />
        </View>
        <View
          dataSet={Object {}}
          style={Object {}}
        />
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onHoverIn={[Function]}
          onHoverOut={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={Object {}}
        >
          <View
            accessibilityState={
              Object {
                "disabled": true,
                "selected": false,
              }
            }
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": "#fecdd3",
                "borderRadius": 9999,
                "marginLeft": 4,
                "marginRight": 4,
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
              }
            }
          />
        </View>
        <View
          dataSet={Object {}}
          style={Object {}}
        />
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onHoverIn={[Function]}
          onHoverOut={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={Object {}}
        >
          <View
            accessibilityState={
              Object {
                "disabled": true,
                "selected": false,
              }
            }
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": "#fecdd3",
                "borderRadius": 9999,
                "marginLeft": 4,
                "marginRight": 4,
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
              }
            }
          />
        </View>
        <View
          dataSet={Object {}}
          style={Object {}}
        />
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onHoverIn={[Function]}
          onHoverOut={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={Object {}}
        >
          <View
            accessibilityState={
              Object {
                "disabled": false,
                "selected": false,
              }
            }
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": "#b5e5fa",
                "borderRadius": 9999,
                "marginLeft": 4,
                "marginRight": 4,
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
              }
            }
          />
        </View>
        <View
          dataSet={Object {}}
          style={Object {}}
        />
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onHoverIn={[Function]}
          onHoverOut={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={Object {}}
        >
          <View
            accessibilityState={
              Object {
                "disabled": false,
                "selected": true,
              }
            }
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 9999,
                "marginLeft": 4,
                "marginRight": 4,
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
              }
            }
          />
        </View>
        <View
          dataSet={Object {}}
          style={Object {}}
        />
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onHoverIn={[Function]}
          onHoverOut={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={Object {}}
        >
          <View
            accessibilityState={
              Object {
                "disabled": false,
                "selected": false,
              }
            }
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": "#b5e5fa",
                "borderRadius": 9999,
                "marginLeft": 4,
                "marginRight": 4,
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
              }
            }
          />
        </View>
        <View
          dataSet={Object {}}
          style={Object {}}
        />
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onHoverIn={[Function]}
          onHoverOut={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={Object {}}
        >
          <View
            accessibilityState={
              Object {
                "disabled": false,
                "selected": false,
              }
            }
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": "#b5e5fa",
                "borderRadius": 9999,
                "marginLeft": 4,
                "marginRight": 4,
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
              }
            }
          />
        </View>
        <View
          dataSet={Object {}}
          style={Object {}}
        />
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onHoverIn={[Function]}
          onHoverOut={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={Object {}}
        >
          <View
            accessibilityState={
              Object {
                "disabled": false,
                "selected": false,
              }
            }
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": "#b5e5fa",
                "borderRadius": 9999,
                "marginLeft": 4,
                "marginRight": 4,
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
